### ISSUE TITLE
List - Keyboard navigation does not work when a `calcite-list` is nested in a `calcite-popover` or `calcite-modal`

### ISSUE THREAD

**Issue by jmanke** [Esri/calcite-components#5968]
### Actual Behavior

Unable to use keyboard (arrow keys) to navigate between items when `calcite-list` is nested in a `calcite-popover` or `calcite-modal`.

### Expected Behavior

Should be able to use keyboard to navigate between items when a `calcite-list` is nested in a `calcite-popover` or `calcite-modal` .

### Reproduction Sample

https://codepen.io/jmanke/pen/PoaRBrq?editors=1000

### Reproduction Steps

1. Activate popover or modal by clicking the 'Activate Popover' or 'Activate Modal' button.
2. Use arrow keys to navigate between list items.
3. Notice keyboard navigation does not work

### Reproduction Version

next

### Relevant Info

Tested in Chrome 108

### Impact

Blocking - we use `calcite-list` nested within a `calcite-popover` and/or `calcite-modal` extensively in Dashboards and keyboard navigation is necessary.

### Esri team

ArcGIS Dashboard

**Comment by jmanke** [comment]
Looks to be fixed on next branch, closing
